### PR TITLE
Added Array Rotation in Every Language Article

### PR DESCRIPTION
**Pull request by agasheaditya**
I am Pulling over here https://github.com/TheRenegadeCoder/sample-programs/pull/2025#issuecomment-705246407
please refer to this PR conversation.

### CODE REVIEW

File: projects/Linked List/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
```
Comment (jrg94) [inline projects/Linked List/README.md:1]: Good start on this file! This will be the only file we work out of.

File: projects/Linked List/linkedlist.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+##a= [-9, -2, 0, 2, 3]
```
Comment (jrg94) [inline projects/Linked List/linkedlist.py:1]: You can remove this file.

File: projects/Linked List/max_sum_arr_rot.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+def max_sum(arr):
```
Comment (jrg94) [inline projects/Linked List/max_sum_arr_rot.py:1]: And, remove this file.

**Review by jrg94 — CHANGES_REQUESTED** [review]
Good start! Let's clean things up a bit.

File: projects/Linked List/README.md (hunk 1 above)
Comment (agasheaditya) [reply projects/Linked List/README.md:1]: okay, I'll remove those, so here we do not want the solutions. am I correct?

File: projects/Linked List/max_sum_arr_rot.py (hunk 3 above)
Comment (agasheaditya) [reply projects/Linked List/max_sum_arr_rot.py]: Done. Please check. thanks man.

**Comment by agasheaditya** [comment]
hey @jrg94, I have submitted the changes but here it shows like some changes are requested. Is there anything which I need to ?
Thanks.
image

**Comment by agasheaditya** [comment]
> I think we need to simplify this documentation, so it only describes one operation. In other words, make this the array rotation pull request. Then, we can work from there.

from the same repo? 
should I create a separate folder for that or the readme for that?

**Comment by jrg94** [comment]
I think this should only be one project: `/projects/array-rotation/index.md`.

Then, we can talk through the details.

**Comment by agasheaditya** [comment]
@jrg94 Please check the project directory. I have made the changes as you have suggested. Thank you.

**Review by jrg94 — CHANGES_REQUESTED** [review]
Looking better! Can you remove on of these files for now? Or open another pull request for it?

From a path naming convention, `projects/array-rotation/index.md`

**Comment by agasheaditya** [comment]
> Looking better! Can you remove on of these files for now? Or open another pull request for it?
> 
> From a path naming convention, `projects/array-rotation/index.md`

hey, I am unable to create a new pull request for the `projects/array-rotation/index.md`. and which one should I delete/remove?like there are only 2 folders I created which are `projects/Linked-list/readme.md` and `projects/array-rotation/readme.md`.
Thanks

**Comment by jrg94** [comment]
Let's stick with array rotation for now.

**Comment by agasheaditya** [comment]
@jrg94 please check now. I removed the `project/linked-list/readme.md`. Thanks.

File: projects/Array-Rotation/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+---
+title: List/Array rotation in Python
```
Comment (jrg94) [inline projects/Array-Rotation/README.md:2]:
```suggestion
title: Array Rotation in Every Language
```

File: projects/Array-Rotation/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+layout: default
+date: 2020-10-14
+last-modified: 2020-10-14
+featured-image: {FEATURED IMAGE NAME}
```
Comment (jrg94) [inline projects/Array-Rotation/README.md:6]:
```suggestion
featured-image: 
```

If you want, you can put an image here. Just make sure to add one to assets/sources.

File: projects/Array-Rotation/README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+tags: Array/ List
+authors:
+  - {AUTHOR USERNAME FROM _data/authors.yml}
```
Comment (jrg94) [inline projects/Array-Rotation/README.md:9]: Make sure you add yourself to this file.

File: projects/Array-Rotation/README.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
+---
+title: List/Array rotation in Python
+layout: default
+date: 2020-10-14
+last-modified: 2020-10-14
+featured-image: {FEATURED IMAGE NAME}
+tags: Array/ List
```
Comment (jrg94) [inline projects/Array-Rotation/README.md:7]:
```suggestion
tags: [array-rotation]
```

File: projects/Array-Rotation/README.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+```shell
+./sample-program.lang sample_input
+```
+
+## Testing
+
+| Description | {EXAMPLE INPUT COLUMN} | Output             |
+| ----------- | ---------------------- | ------------------ |
+| List/ Array |    input =[8,3,1,2]    | Mximum Sum is :29  |
```
Comment (jrg94) [inline projects/Array-Rotation/README.md:53]: We'll need several tests. Take a look at some examples in the project list to get started.

File: projects/Array-Rotation/README.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+
+## Articles
+
+{% include article_list.md collection=site.categories.{PROJECT NAME} %}
```
Comment (jrg94) [inline projects/Array-Rotation/README.md:58]:
```suggestion
{% include article_list.md collection=site.categories.array-rotation %}
```

**Review by jrg94 — CHANGES_REQUESTED** [review]
Made a few quick changes. Take some time to compare this file with some of the others in the project list.

File: projects/Array-Rotation/README.md (hunk 6 above)
Comment (agasheaditya) [reply projects/Array-Rotation/README.md:9]: hey, whereto find that authors.yml ? I didn't find one in the repo.

File: projects/Array-Rotation/README.md (hunk 8 above)
Comment (agasheaditya) [reply projects/Array-Rotation/README.md:53]: okay 
I will add some more

File: projects/Array-Rotation/README.md (hunk 6 above)
Comment (jrg94) [reply projects/Array-Rotation/README.md:9]: The path is in that line: `_data/authors.yml`

**Comment by agasheaditya** [comment]
hey @jrg94, I made those changes. added some test cases to the readme and also added my self to authors.yml. please check
thanks

File: projects/Array-Rotation/README.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+##### Python 3.* 
+
+```shell
+./sample-program.lang sample_input
+```
+
+## Testing
+
+| Description | {EXAMPLE INPUT COLUMN} | Output             |
```
Comment (jrg94) [inline projects/Array-Rotation/README.md:51]:
```suggestion
| Description | Input | Output             |
```

File: projects/Array-Rotation/README.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+| ----------- | ---------------------- | ------------------ |
+| List/ Array |    input =[8,3,1,2]    | Maximum Sum is:29  |
```
Comment (jrg94) [inline projects/Array-Rotation/README.md:53]:
```suggestion
| List/ Array | "8,3,1,2" | 29 |
```

File: projects/Array-Rotation/README.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+| List/ Array |  input = [15,18,20,70] | Maximum Sum is: 268|
+| List/ Array |ip= [24,4,7,-8,6,77,14] | Maximum Sum is: 531|
+
+## Articles
+
+{% include article_list.md collection=site.categories.Array-rotation %}
```
Comment (jrg94) [inline projects/Array-Rotation/README.md:59]:
```suggestion
{% include article_list.md collection=site.categories.array-rotation %}
```

**Review by jrg94 — CHANGES_REQUESTED** [review]
We're getting a bit closer! Take another peek at the suggestions for testing. Ideally, the user would run the program like this:

```
array-rotation.lang "1, 4, 3, 2"
```

And, it would output some number directly.

**Comment by agasheaditya** [comment]
> We're getting a bit closer! Take another peek at the suggestions for testing. Ideally, the user would run the program like this:
> 
> ```
> array-rotation.lang "1, 4, 3, 2"
> ```
> 
> And, it would output some number directly. 

Okay. Means I need to use argparse for the same? I'm I correct? Or should I add the "1,4,3,2" test case ?

**Comment by jrg94** [comment]
I'm not sure what you mean, but the example is how I would expect this program to work. I'll take a quick clean up of the file. Feel free to take a look when I'm done.

**Comment by jrg94** [comment]
Alright, I've updated the testing table. We definitely need to go through and add some error cases. We usually add these: no input & empty string input. Might be worth adding a case for valid numbers, but we don't do that for sorting: https://sample-programs.therenegadecoder.com/projects/bubble-sort/

**Comment by agasheaditya** [comment]
> Alright, I've updated the testing table. We definitely need to go through and add some error cases. We usually add these: no input & empty string input. Might be worth adding a case for valid numbers, but we don't do that for sorting: https://sample-programs.therenegadecoder.com/projects/bubble-sort/

will do

**Comment by jrg94** [comment]
Don't forget to resolve the merge conflict as well! 👍

**Comment by agasheaditya** [comment]
@jrg94 I have added some 2 more test cases for empty input and no input, also resolved the conflict issue. Thanks.

**Review by jrg94 — APPROVED** [review]
Alright, I reworked this a bit! I think this is in good shape now.

**Comment by agasheaditya** [comment]
> Alright, I reworked this a bit! I think this is in good shape now.

thank you very much @jrg94. It was great working with you.
